src/EduManagementLab.Web/Pages/Courses/Create.cshtml.cs

```diff
@@ -37,14 +38,20 @@ public IActionResult OnGet()
         // To protect from overposting attacks, see https://aka.ms/RazorPagesCRUD
         public async Task<IActionResult> OnPostAsync()
         {
-            if (!ModelState.IsValid)
+            if (ModelState.IsValid)
             {
-                return Page();
+                try
+                {
+                    _courseService.CreateCourse(Code, Name, Description, StartDate, EndDate);
+                    return RedirectToPage("./Index");
+                }
+                catch (Exception)
+                {
+                    ModelState.AddModelError(String.Empty, "Course code/name already exist");
```

Comment:
You dont know that the Exception is thrown because that code/name already exists.
It can be another error.